client/src/data-access/userApi.ts

```diff
@@ -46,23 +49,29 @@ export class UserApi {
             lastName,
             age: 30,
             email,
-            password
+            password,
         })
     }
 
-    public updateUserPassword = async (
-       payload: IChangePassword
-    ) => {
-        return this.api.post('/auth/reset', payload);
+    public updateUserPassword = async (payload: IChangePassword) => {
+        return this.api.post('/auth/reset', payload)
     }
 
     public getCurrentUser = async (token: string) => {
-        return this.api.get('/users/me',{ headers: {authorization: token}})
+        return this.api.get('/users/me', { headers: { authorization: token } })
     }
 
     public signInWithProvider = async (providerName: providerType) => {
-        const result = await this.authService.signInWithPopup(providerName)
-        return result
+        try {
+            const result = await hello(providerName).login()
+
+            return this.api.post('/auth/provider', {
```

Comment:
It would make sense to have provider as part of the url and not body. Eg /auth/facebook and /auth/google, auth/twitter. On the server side use placeholders, eg /auth/:providerName